apps/cpu-profile-summarizer/src/start.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { once } from 'node:events';
+import fs from 'node:fs';
+import os from 'node:os';
+import path from 'node:path';
+import { parseArgs } from 'node:util';
+import type { Worker } from 'node:worker_threads';
+
+import { WorkerPool } from '@rushstack/worker-pool';
+
+import type { IMessageFromWorker } from './protocol';
+import type { INodeSummary, IProfileSummary } from './types';
+
+/**
+ * Merges summarized information from multiple profiles into a single collection.
+ * @param accumulator - The collection to merge the nodes into
+ * @param values - The nodes to merge
+ */
+function mergeProfileSummaries(
+  accumulator: Map<string, INodeSummary>,
+  values: Iterable<[string, INodeSummary]>
+): void {
+  for (const [nodeId, node] of values) {
+    const existing: INodeSummary | undefined = accumulator.get(nodeId);
+    if (!existing) {
+      accumulator.set(nodeId, node);
+    } else {
+      existing.selfTime += node.selfTime;
+      existing.totalTime += node.totalTime;
+    }
+  }
+}
+
+/**
+ * Scans a directory and its subdirectories for CPU profiles.
+ * @param baseDir - The directory to recursively search for CPU profiles
+ * @returns All .cpuprofile files found in the directory and its subdirectories
+ */
+function findProfiles(baseDir: string): string[] {
+  baseDir = path.resolve(baseDir);
+
+  const files: string[] = [];
+  const directories: string[] = [baseDir];
+
+  for (const dir of directories) {
+    const entries: fs.Dirent[] = fs.readdirSync(dir, { withFileTypes: true });
+    for (const entry of entries) {
+      if (entry.isFile() && entry.name.endsWith('.cpuprofile')) {
+        files.push(`${dir}/${entry.name}`);
+      } else if (entry.isDirectory()) {
+        directories.push(`${dir}/${entry.name}`);
+      }
+    }
+  }
+
+  return files;
+}
+
+/**
+ * Prints the usage information for the application.
+ */
+function printUsage(): void {
+  console.log('Usage: cpu-profile-summarizer --input DIR [[--input DIR] ...] --output TSV_FILE');
+}
+
+/**
+ * Processes a set of CPU profiles and aggregates the results.
+ * Uses a worker pool.
+ * @param profiles - The set of .cpuprofile files to process
+ * @returns A summary of the profiles
+ */
+async function processProfilesAsync(profiles: Set<string>): Promise<IProfileSummary> {
+  const maxWorkers: number = Math.min(profiles.size, os.availableParallelism());
+  console.log(`Processing ${profiles.size} profiles using ${maxWorkers} workers...`);
+  const workerPool: WorkerPool = new WorkerPool({
```

Comment:
I'm curious what the benefit of using a workerpool vs multiple individual processes provides. Is it mostly avoiding the overhead of Node spinning up?